### PR TITLE
Add focus trap and shortcut for fullscreen.

### PR DESCRIPTION
**Pull request by dbatiste**
* adds focus trap while in fullscreen
* adds shortcut for toggling fullscreen (Cmd+Shift+F / Ctrl+Shift+F)
* limit the width in fullscreen to 1230px as per design

### CODE REVIEW

File: components/plugins/attributes.js [hunk 1]
```diff
@@ -19,7 +19,8 @@ const name = 'd2l-attributes';
 export function addAttributesPlugin() {
 	tinymce.PluginManager.add(name, function(editor) {
 
-		const root = editor.getElement().getRootNode();
+		const context = requestInstance(editor.getElement(), 'context');
```
Comment (dbatiste) [inline components/plugins/attributes.js:22]: I originally made this change to support the idea that I would need to manage the placement of the dialogs within the focus trap, but so far it has not been necessary.  Nevertheless, I would like to tidy up method for getting the component, as well as the `requestInstance` stuff so I decided to keep these changes.

File: components/plugins/fullscreen.js [hunk 2]
```diff
@@ -1,28 +1,43 @@
...
+		const context = requestInstance(editor.getElement(), 'context');
+		const component = context.component;
 
-		editor.addQueryStateHandler(cmds.fullscreen, () => {
-			return state;
-		});
+		let state = false;
 
-		editor.addCommand(cmds.fullscreen, () => {
+		const toggleFullscreen = () => {
```
Comment (dbatiste) [inline components/plugins/fullscreen.js:14]: This is the same logic as before - it's just been factored out so that it can be invoked both when the the user clicks the fullscreen toggle button, or when they use the shortcut key.

File: htmleditor-mixin.js [hunk 3]
```diff
@@ -228,6 +228,10 @@ export const HtmlEditorMixin = superclass => class extends Localizer(RtlMixin(Pr
 		this.provideInstance('wordCountInFooter', this.wordCountInFooter);
 		this.provideInstance('localize', this.localize.bind(this));
 
+		this.provideInstance('context', { ...{
```
Comment (dbatiste) [inline htmleditor-mixin.js:231]: As mentioned in my previous comment I would like to tidy this stuff up to make it less chatty. So my plan it to move these other items into this context and spread the private context into this as well.  I figured I would leave that for another PR as to not confuse this PR.

File: components/plugins/fullscreen.js [hunk 4]
```diff
@@ -1,28 +1,43 @@
...
+			if (e.keyCode !== 70) return;
+			if (!(e.metaKey && e.shiftKey) && !(e.ctrlKey && e.shiftKey)) return;
+			toggleFullscreen();
+		};
+
+		editor.addQueryStateHandler(cmds.fullscreen, () => state);
+
+		editor.addCommand(cmds.fullscreen, toggleFullscreen);
+
+		// inside iframe (note: metaKey combination only works on keydown)
```
Comment (grant-cleary) [inline components/plugins/fullscreen.js:36]: Interesting. I didn't know that about metaKey.

File: htmleditor-mixin.js (hunk 3 above)
Comment (grant-cleary) [reply htmleditor-mixin.js:231]: Makes sense to me.

**Review by grant-cleary — APPROVED** [review]

File: components/plugins/fullscreen.js (hunk 4 above)
Comment (dbatiste) [reply components/plugins/fullscreen.js:36]: Yeah, on Mac it is the Cmd (or otherwise know as the 🍎 key), and I think on Windows is can be the Windows key.  Though I think there can be some inconsistency with it.  This logic however, doesn't do any OS detection, so on Mac, either combination will work.  Hopefully that does not cause any conflicts.  The Shift key effectively mitigates the conflict with the standard Cmd/Ctrl+F for "Find".

Comment (grant-cleary) [reply components/plugins/fullscreen.js:36]: Cool yeah, makes sense to me.
